src/main/java/entities/Game.java

```diff
@@ -0,0 +1,164 @@
+package entities;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Game implements ObservableGame {
+    private Card currentCard;
+    private ArrayList<Player> players;
+
+    private Deck gameDeck;
+
+    private Player winner;
+
+    private Player currentTurn;
+
+    private ArrayList<GameObserver> observers;
+
+    /**
+     * Construct a new card game given a particular deck and set of players.
+     * @param deck The deck of cards that belongs to this Game.
+     * @param players The Players who will be playing this card Game.
+     */
+    public Game(Deck deck, ArrayList<Player> players) {
+        this.gameDeck = deck;
+        this.players = new ArrayList<>(players);
+        this.currentTurn = this.players.get(0);
+    }
+
+    /**
+     * Get the Player whose turn it is currently.
+     * @return The Player in this Game whose turn it is who needs to put down a card.
+     */
+    public Player getCurrentTurn() {
+        return this.currentTurn;
+    }
+
+    /**
+     * Change the turn of the Player from the current Player to the next in the Game.
+     * This method assumes that the Player before is not the winner of the Game, otherwise it would not be called.
+     */
+    private void changeCurrentTurn() {
+        int currentPlayerIndex = this.players.indexOf(this.currentTurn);
+        if(currentPlayerIndex == (this.players.size() -  1)) {
+            this.currentTurn = this.players.get(0);
+        }
+        else {
+            this.currentTurn = this.players.get(currentPlayerIndex + 1);
+        }
+    }
+
+    /**
+     * Get the Card that had been played last.
+     * @return A Card object which the current Player must place their card on top of.
+     */
+    public Card getCurrentCard() {
+        return this.currentCard;
+    }
+
+    /**
+     * A Player will play the given card object to be put down into the Game as the new currentCard.
+     * @param card A Card object played by a Player in this Game.
+     */
+    public void putCardDown(Card card) {
+        if (currentCard != null) {
+            this.gameDeck.addCardToDeck(this.currentCard);
+        }
+        this.currentCard = card;
+    }
+
+    /**
+     * Set the winner of this game to the given Player.
+     * @param winner The Player who has won the Game.
+     */
+    public void setWinner(Player winner) {
+        this.winner = winner;
+    }
+
+    /**
+     * Get all the Players in this Game.
+     * @return an ArrayList of Players in this game.
+     */
+    public ArrayList<Player> getPlayers() {
```

Comment:
should specify a copy of the list of players is returned

Reply:
Done